api/server/runner.go

```
@@ -32,12 +39,15 @@ func (s *Server) handleFunctionCall(c *gin.Context) {
 	var a string
 	ai := ctx.Value(api.AppName)
 	if ai == nil {
-		handleErrorResponse(c, errors.New("app name not set"))
-		return
+		err := models.ErrAppsMissingName
+		return err
```

Why not just return that error?